packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -237,26 +240,9 @@ export class OdspDocumentService implements IDocumentService {
                 ? Promise.resolve(null)
                 : this.getWebsocketToken!(options);
 
-            const joinSessionPromise = this.joinSession(requestWebsocketTokenFromJoinSession, options).catch((e) => {
-                const likelyFacetCodes = e as IFacetCodes;
-                if (Array.isArray(likelyFacetCodes.facetCodes)) {
-                    for (const code of likelyFacetCodes.facetCodes) {
-                        switch (code) {
-                            case "sessionForbiddenOnPreservedFiles":
-                            case "sessionForbiddenOnModerationEnabledLibrary":
-                            case "sessionForbiddenOnRequireCheckout":
-                                // This document can only be opened in storage-only mode.
-                                // DeltaManager will recognize this error
-                                // and load without a delta stream connection.
-                                this._policies = {...this._policies,storageOnly: true};
-                                throw new DeltaStreamConnectionForbiddenError(code, { driverVersion });
-                            default:
-                                continue;
-                        }
-                    }
-                }
-                throw e;
-            });
+            const refreshJoinSessionPolicy = this.mc.config.getBoolean("Fluid.Driver.Odsp.refreshJoinSession");
```

Comment:
I'd prefer most of gates like that to be kill-bit switches. I.e. it's enabled by default, and we can disable it through ESC if we need to, not the other way around (disabled by default, and ESC is used to roll out).

It's mostly a matter of size. If it's a tiny feature, running experiments adds a lot of friction costs, so enabling by default removes that cost. Big and very risky new features should do the opposite - use ESC for slow roll out

Reply:
Reverted the policy to be a kill switch.